src/topic_monitor.cpp

```diff
@@ -23,7 +23,7 @@ wcstring generation_list_t::describe() const {
     return result;
 }
 
-binary_semaphore_t::binary_semaphore_t() : sem_ok_(false) {
+binary_semaphore_t::binary_semaphore_t() {
```

Comment:
Looking at the surrounding code, I think this initialization helps readability, because there is an 
`if (!sem_ok_)` below. Normally I'd prefer an explicit default value right at the field definition, but here it makes much more sense to place this information at the only constructor.